curator-client/src/main/java/org/apache/curator/utils/Compatibility.java

```diff
@@ -18,29 +18,49 @@
  */
 package org.apache.curator.utils;
 
+import org.apache.zookeeper.WatchedEvent;
+import org.apache.zookeeper.Watcher;
 import org.apache.zookeeper.ZooKeeper;
+import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import java.lang.reflect.Method;
 
 /**
  * Utils to help with ZK 3.4.x compatibility
  */
 public class Compatibility
 {
     private static final boolean hasZooKeeperAdmin;
+    private static final Method queueEventMethod;
+    private static final Logger logger = LoggerFactory.getLogger(Compatibility.class);
+
     static
     {
-        boolean hasIt;
+        boolean localHasZooKeeperAdmin;
         try
         {
             Class.forName("org.apache.zookeeper.admin.ZooKeeperAdmin");
-            hasIt = true;
+            localHasZooKeeperAdmin = true;
         }
         catch ( ClassNotFoundException e )
         {
-            hasIt = false;
-            LoggerFactory.getLogger(Compatibility.class).info("Running in ZooKeeper 3.4.x compatibility mode");
+            localHasZooKeeperAdmin = false;
+            logger.info("Running in ZooKeeper 3.4.x compatibility mode");
+        }
+        hasZooKeeperAdmin = localHasZooKeeperAdmin;
+
+        Method localQueueEventMethod;
+        try
+        {
+            Class<?> testableClass = Class.forName("org.apache.zookeeper.Testable");
+            localQueueEventMethod = testableClass.getMethod("queueEvent", WatchedEvent.class);
```

Comment:
Did you actually want to try calling getMethod("queueEvent", ) on the ZooKeeper class? There is no queueEvent method in Testable, but there is in ZooKeeper of 3.5.